xmake/toolchains/zig/xmake.lua

```diff
@@ -52,24 +52,43 @@ toolchain("zig")
     on_load(function (toolchain)
         import("core.base.semver")
 
+        -- @see https://github.com/xmake-io/xmake/issues/5610
+        function _generate_zigcc_wrapper(zig)
```

Comment:
we should wrap these tools in toolchain/on_check, and save to toolchain:config(), then get it in on_load